x/bsonx/bsoncore/bsoncore_test.go

```diff
@@ -899,6 +900,60 @@ func TestBuild(t *testing.T) {
 	}
 }
 
+func TestNullBytes(t *testing.T) {
+	// Helper function to execute the provided callback and assert that it panics with the expected message. The
+	// createBSONFn callback should create a BSON document/array/value and return the stringified version.
+	assertDocumentCreationPanics := func(t *testing.T, createBSONFn func() string, expected string) {
+		t.Helper()
+
+		var res string
+		defer func() {
+			got := recover()
```

Comment:
Cool, hadn't come across panic recovery yet in Go.

Reply:
Oh neat, this is new to me as well.